### ISSUE TITLE
quit after startup command invocation

### ISSUE THREAD

**Issue by dhait** [python-cmd2/cmd2#452]
How would you use cmd2 to automatically quit after invoking the startup commands; but enter a loop if and only if there are no commands specified on the command line?

so
```
mycmd junk
```
executes the junk command and exits, but
```
mycmd
```
prints the prompt and waits for commands

**Comment by tleonhardt** [comment]
Hi @dhait, there isn't currently a built-in feature for doing this exactly as you wish.  But you can do the following if you wish to quit after running a command and not enter a loop:
```Bash
mycmd junk quit
```

Would that meet your needs?

Since you are not the first person to want this, I would consider adding a feature to optionally support this mode if we can think of a nice clean way to do it.  I know someone else posted a code example for doing what you are talking about, I will see if I can dig up a link ...

**Comment by kotfu** [comment]
@dhait, I maintain a `cmd2` based application which does exactly what you describe. The relevant code is shown in https://github.com/tomcatmanager/tomcatmanager/blob/master/tomcatmanager/__main__.py.

There is a big more going on in there than you have asked for, including using `argparse` to process command line arguments from the shell invocation. It also checks the results of the `cmd2` command and sets a shell exit code, which you may or may not want to do.

This code is in ``__main.py__`` because `tomcatmanager` uses a feature of `setuptools` to install command line executives as described at http://python-packaging.readthedocs.io/en/latest/command-line-scripts.html

The resulting behavior is just what you requested:
```
$ tomcatmanager help
```
executes the `help` command and exits, while
```
$ tomcatmanager
```
enters the interactive command loop.

**Comment by tleonhardt** [comment]
Lets leave this issue open and add an example demonstrating how to do this and possibly add something to the documentation as well since this appears to be a common use-case.

**Comment by dhait** [comment]
I also wonder if you can't hack something into `preloop` to check the args and push a quit command on the `cmdqueue` if called for....

**Comment by tleonhardt** [comment]
If we did decide to hack something in to support this it would probably be a boolean flag attribute with a name like ``quite_after_cli_args`` which if ``True`` would convert from the default behavior of entering the REPL after executing the commands to quitting after executing the commands.

Would that be significantly easier for end users than just creating a ``__main__`` like I showed above?

**Comment by dhait** [comment]
It's conceivable that in addition to "commands" being listed as args, you might also have "options".
So for example,
```
mycmd --foo test install
```
might execute the "test" and "install" commands, with the option "foo" set, and then exit.  In this case, the simple `__main__` you gave above would need to be modified to handle the optional args.  I suppose it's also possible the optional args could come AFTER the commands.

**Comment by tleonhardt** [comment]
Arguments to commands issued at invocation are currently supported out-of-the-box, you just need to put each set of command+args in quotes so that ``cmd2`` knows how to unambiguously parse everything:
```Bash
14:40 $ ./main.py "help -v" 'set -l'

Documented commands (type help <topic>):
================================================================================
alias               Define or display aliases
edit                Edit a file in a text editor.
help                List available commands with "help" or detailed help with "help cmd".
history             View, run, edit, and save previously entered commands.
ipy                 Enters an interactive IPython shell.
load                Runs commands in script file that is encoded as either ASCII or UTF-8 text.
py                  Invoke python command, shell, or script
pyscript            Runs a python script file inside the console
quit                Exits this application.
set                 Sets a settable parameter or shows current settings of parameters.
shell               Execute a command as if at the OS prompt.
shortcuts           Lists shortcuts (aliases) available.
unalias             Unsets aliases

colors: True              # Colorized output (*nix only)
continuation_prompt: >    # On 2nd+ line of input
debug: False              # Show full error stack on error
echo: False               # Echo command issued into output
editor: vim               # Program used by ``edit``
feedback_to_output: False # Include nonessentials in `|`, `>` results
locals_in_py: True        # Allow access to your application in py via self
prompt: (Cmd)             # The prompt issued to solicit input
quiet: False              # Don't print nonessential feedback
timing: False             # Report execution times
(Cmd)
```

**Comment by tleonhardt** [comment]
@kmvanbrunt @kotfu @anselor Do you think we should add a boolean flag to allow developers to alter the default behavior when extra arguments are provided on the command line when running a ``cmd2`` application?  

Or should we just add an example and a mention of that example in the documentation?

**Comment by kotfu** [comment]
I'm not in favor of adding a new boolean flag for this one. I think we have shown that there are several ways to accomplish the task without modifying cmd2. I do think an example and a mention in the documentation would be a great idea.

**Comment by kotfu** [comment]
Another option is to accept commands from standard input, one command per line of input. This allows you to use the command line switches unambiguously:

```
$ mycmd --foo <<EOF
> junk --with --switches
> set -l
> quit
> EOF
```

**Comment by tleonhardt** [comment]
This was closed by #482
